src/List/Extra.elm

```diff
@@ -1244,6 +1244,74 @@ indexedFoldr func acc list =
     second (List.foldr step ( List.length list - 1, acc ) list)
 
 
+{-| A custom type used for stoppable folds.
+
+    stoppableFoldl
+        (\n acc ->
+            if acc >= 50 then
+                Stop acc
+
+            else
+                Continue (n + acc)
+        )
+        0
+        (List.range 1 10000)
+    --> 55
+
+-}
+type Step a
+    = Continue a
+    | Stop a
+
+
+{-| A `foldl` that can stop early instead of traversing the whole list.
+
+    stoppableFoldl
+        (\n acc ->
+            if acc >= 50 then
+                Stop acc
+            else
+                Continue (n + acc)
+        )
+        0
+        (List.range 1 10000)
+    --> 55
+
+-}
+stoppableFoldl : (a -> b -> Step b) -> b -> List a -> b
+stoppableFoldl func acc list =
+    case list of
+        [] ->
+            acc
+
+        x :: xs ->
+            case func x acc of
+                Continue newAcc ->
+                    stoppableFoldl func newAcc xs
+
+                Stop finalAcc ->
+                    finalAcc
+
+
+{-| A `foldr` that can stop early instead of traversing the whole list.
```

Comment:
But it does traverse the whole list because of the `reverse` :smiling_face_with_tear:

Reply:
I don't think there are a lot of performance comments in this package, but it may be worth mentioning that if the whole list ends up being visited, that this version will be slower than `List.foldr`. What do you think?

Reply:
I think both are possible solutions. Not providing foldr is definitely the easiest one.